test/unit/Types/ScalarTypesResolverTest.php

```diff
@@ -0,0 +1,210 @@
+        Assert::assertEquals($expectedValue, $convertedValue);
+    }
+
+    /**
+     * @return array|mixed[]
+     *
+     * @throws DatetimeException
+     */
+    public function convertSerializeDataProvider(): array
+    {
+        return [
+            'serialize_null' => [
+                'deserialize' => false,
+                'id' => 1,
+                'value' => null,
+                'expectedValue' => null,
+            ],
+            'serialize_date' => [
+                'deserialize' => false,
+                'id' => 1,
+                'value' => DateTime::createFromFormat('Y-m-d', '2020-07-31'),
+                'expectedValue' => '2020-07-31',
+            ],
+            'serialize_date_time' => [
+                'deserialize' => false,
+                'id' => 2,
+                'value' => new DateTime('2020-07-31 15:35:11'),
+                'expectedValue' => '2020-07-31T15:35:11+00:00',
+            ],
+            'serialize_byte' => [
+                'deserialize' => false,
+                'id' => 3,
+                'value' => 'This is an encoded string',
+                'expectedValue' => 'VGhpcyBpcyBhbiBlbmNvZGVkIHN0cmluZw==',
+            ],
+            'serialize_value_without_serializer_function' => [
+                'deserialize' => false,
+                'id' => 11,
+                'value' => 'q1werty',
+                'expectedValue' => 'q1werty',
+            ],
+        ];
+    }
+
+    /**
+     * @param mixed $value
+     * @param mixed $expectedValue
+     *
+     * @dataProvider convertSerializeDataProvider
+     */
+    public function testConvertSerializesValue(bool $deserialize, int $id, $value, $expectedValue): void
+    {
+        $convertedValue = $this->scalarTypeResolver->convert($deserialize, $id, $value);
+        Assert::assertEquals($expectedValue, $convertedValue);
+    }
+
+    public function testGetPatternReturnsNull(): void
+    {
+        $pattern = $this->scalarTypeResolver->getPattern(0);
+        Assert::assertNull($pattern);
+    }
+
+    public function testGetPatternReturnsPattern(): void
+    {
+        $pattern         = $this->scalarTypeResolver->getPattern(1);
+        $expectedPattern = '\d{4}-\d{2}-\d{2}';
+        Assert::assertSame($expectedPattern, $pattern);
+    }
+
+    public function testGetPhpTypeReturnsPhpType(): void
+    {
+        $phpType         = $this->scalarTypeResolver->getPhpType(0);
+        $expectedPhpType = 'string';
+        Assert::assertSame($expectedPhpType, $phpType);
+    }
+
+    public function testIsDateTimeReturnsTrue(): void
+    {
+        $isDateTime = $this->scalarTypeResolver->isDateTime(0);
+        Assert::assertFalse($isDateTime);
+    }
+
+    public function testIsDateTimeReturnsFalse(): void
+    {
+        $isDateTime = $this->scalarTypeResolver->isDateTime(1);
+        Assert::assertTrue($isDateTime);
+    }
+
+    public function testFindScalarTypeTypeIsNullReturnsZero(): void
```

Comment:
I propose to rename the methods ending with "ReturnsZero" to "ReturnsStringTypeId".
The logic here is that for types without a format or unknown types the bundle should serialize them as a string, the test should reflect that too.